src/bridge-canister/src/runtime/service/mint_tx.rs

```diff
@@ -0,0 +1,144 @@
+    handler: H,
+    orders_to_send: RefCell<HashMap<H256, MintOrderBatchInfo>>,
+}
+
+impl<H> SendMintTxService<H> {
+    /// Creates a new service with the given handler.
+    pub fn new(handler: H) -> Self {
+        Self {
+            handler,
+            orders_to_send: Default::default(),
+        }
+    }
+}
+
+#[async_trait::async_trait(?Send)]
+impl<H: MintTxHandler> BridgeService for SendMintTxService<H> {
+    async fn run(&self) -> BftResult<()> {
+        log::trace!("Running SendMintTxService");
+
+        let Some((digest, batch_info)) = self
+            .orders_to_send
+            .borrow()
+            .iter()
+            .map(|(digest, batch_info)| (digest.clone(), batch_info.clone()))
+            .next()
+        else {
+            log::trace!("No mint orders batch ready to be sent.");
+            return Ok(());
+        };
+
+        let config = self.handler.get_evm_config();
+
+        let signer = config.borrow().get_signer()?;
+        let sender = signer.get_address().await?;
+
+        let bridge_contract =
+            config
+                .borrow()
+                .get_bft_bridge_contract()
+                .ok_or(Error::Initialization(
+                    "Singing service failed to get BftBridge address".into(),
+                ))?;
+
+        let evm_params = config.borrow().get_evm_params()?;
+        let tx_params = evm_params.create_tx_params(sender, bridge_contract);
+
+        log::trace!(
+            "Sending batchMint transaction with {} mint orders.",
+            batch_info.orders_batch.orders_number()
+        );
+        let mut tx = bft_events::batch_mint_transaction(
+            tx_params,
+            &batch_info.orders_batch.orders_data,
+            &batch_info.orders_batch.signature,
+            &[],
+        );
+
+        let signature = signer.sign_transaction(&(&tx).into()).await?;
+        tx.r = signature.r.0;
+        tx.s = signature.s.0;
+        tx.v = signature.v.0;
+        tx.hash = tx.hash();
+
+        let client = config.borrow().get_evm_link().get_json_rpc_client();
+        let tx_hash = client.send_raw_transaction(tx).await.map_err(|e| {
+            Error::EvmRequestFailed(format!("failed to send batch mint tx to EVM: {e}"))
+        })?;
+
+        log::trace!(
+            "The batchMint transaction with {} mint orders sent.",
+            batch_info.orders_batch.orders_number()
+        );
+
+        // Remove sent orders batch from service.
+        let sent_batch_info = match self.orders_to_send.borrow_mut().remove(&digest) {
+            Some(batch_info) => batch_info,
+            None => {
+                log::warn!("Failed to remove signed mint orders which was just sent.");
+                batch_info
+            }
+        };
+
+        // Update state for all operations related with the orders batch.
+        for op_id in sent_batch_info.related_operations {
+            self.handler.mint_tx_sent(op_id, tx_hash.into())
```

Comment:
> 1. to add the index of mint order in the batch to the operation state, so that I don't need to decode tx inputs.
Index is a part of Operation state:
```rust
pub struct SignedOrder {
    all_orders: SignedOrders,
    idx: OrderIdx,
}
```

> 2. to return at least some kind of status code instead of just success/fail for each mint order.
I agree. I'll replace the output booleans with error code.